distros/ros2-overlay.nix

```diff
@@ -248,6 +248,22 @@ rosSelf: rosSuper: with rosSelf.lib; {
     };
   });
 
+  rviz-ogre-vendor = rosSuper.rviz-ogre-vendor.overrideAttrs ({
+    nativeBuildInputs ? [], ...
+  }: {
+    nativeBuildInputs = nativeBuildInputs ++ optionals self.stdenv.isDarwin [
+      self.darwin.apple_sdk.frameworks.Foundation
+      self.darwin.apple_sdk.frameworks.AppKit
+    ];
+  });
+
+  # lttng-ust and lttng-tools is not available on Darwin, will disable tracking if not available.
+  tracetools = rosSuper.tracetools.overrideAttrs ({
+    propagatedBuildInputs ? [], ...
+  }: {
+    propagatedBuildInputs = if self.stdenv.isDarwin then [] else propagatedBuildInputs;
```

Comment:
Can you implement this the same way as in `rmw-implementation`? That is:
```nix
    buildInputs = if self.stdenv.isDarwin then
      builtins.filter (p: p.pname != "ros-${p.rosDistro}-rmw-cyclonedds-cpp") buildInputs
    else
      buildInputs;
```
This way, when more packages is later added to propagatedBuildInputs of tracetools, those won't be removed.